src/shared/nodeTypes.ts

```diff
@@ -65,7 +65,7 @@ export type NodeWithDefinedTypeArguments = ts.Node & {
 // TODO: make this a more specific type
 // Will have to deal with instantiations (new Container<T>() { ... }) and declarations (class Container<T>() { ... }))
 export type NodeWithDefinedTypeParameters = ts.Node & {
```

Comment:
[Naming] Nit: the original intent for this was that it's a "node" whose "type parameters" are _defined_. But now it's just a node with the property.

```suggestion
export type NodeWithTypeParameters = ts.Node & {
```

(here and the `isNodeWithDefinedTypeParameters` function too)